prov/sockets/include/sock.h

```diff
@@ -1124,11 +1129,9 @@ int sock_conn_start_listener_thread(struct sock_conn_listener *conn_listener);
 int sock_conn_stop_listener_thread(struct sock_conn_listener *conn_listener);
 void sock_conn_map_destroy(struct sock_ep_attr *ep_attr);
 void sock_conn_release_entry(struct sock_conn_map *map, struct sock_conn *conn);
-void sock_set_sockopts(int sock);
+void sock_set_sockopts(int sock, enum sock_opts);
```

Comment:
Flags are passed in here, not enum values.  Convert to int.  E.g. NONBLOCK | KEEPALIVE produce a value that is not a value of the enum, which will produce compiler warnings/errros.